packages/flutter/lib/src/material/tooltip.dart

```diff
@@ -30,6 +30,59 @@ import 'tooltip_theme.dart';
 ///
 /// {@youtube 560 315 https://www.youtube.com/watch?v=EeEfD5fI-5Q}
 ///
+/// ## Example
+///
+/// {@tool dartpad --template=stateless_widget_material}
+///
+/// This example show a basic [Tooltip] which has a [Text] as child.
+/// [message] contains your label to be shown on long pressed.
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return Center(
+///     child: Tooltip(
+///       message: "Welcome Greetings",
+///       child: Text("Hello Flutter"),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// {@tool dartpad --template=stateless_widget_material}
+///
+/// This example covers most of the attributes available in Tooltip.
+/// `decoration` has been used to give a gradient and borderRadius to Tooltip.
+/// `height` has been used to set a specific size of the Tooltip.
+/// `preferBelow` accepts two values true or false, true will show tooltip below
+/// the child and false will show tooltip above child.
+/// `textStyle` has been used to set fontSize of 'message'.
+/// `showDuration` accepts a Duration to show the message after long Press released.
+/// `waitDuration` accepts a Duration to show the message after Long Press initiated.
```

Comment:
This is actually for mouse pointers instead of long presses: https://master-api.flutter.dev/flutter/material/Tooltip/waitDuration.html

```suggestion
/// `waitDuration` accepts a Duration for which a mouse pointer has to hover over the child
/// widget before the tooltip is shown.
```